crates/p2p_proto/proto/snapshot.proto

```diff
@@ -0,0 +1,84 @@
+syntax = "proto3";
+
+package starknet.snapshot;
+
+import "common.proto";
+import "state.proto";
+
+message PatriciaNode {
+    message Edge {
+        uint32 length = 1;
+        starknet.common.Felt252 path = 2; // as bits of left/right
+        starknet.common.Felt252 value = 3;
+    }
+    message Binary {
+        starknet.common.Felt252 left = 1;
+        starknet.common.Felt252 right = 2;
+    }
+
+    oneof node {
+        Edge edge = 1;
+        Binary binary = 2;
+    }
+}
+
+// non leaf nodes required to build the trie given the range (leaves)
+message PatriciaRangeProof {
+    repeated PatriciaNode nodes = 1;
+}
+
+// leafs of the contract state tre
+message ContractState {
+    starknet.common.Hash address = 1; // the key
+    starknet.common.Hash class = 2;
+    starknet.common.Hash storage = 3; // patricia
+    uint64 nonce = 4;
+}
+
+// request a range from the contract state tree that matches the given root (block)
+// starts at 'start' and ends no less than 'end'.
+// the result is  (ContractRange+, PatriciaRangeProof)*
+message GetContractRange {
+    starknet.common.Hash root = 2;
+    starknet.common.Hash start = 3;
+    starknet.common.Hash end = 4;
+    uint32 chunks_per_proof = 5; // how many ContractRange items to send before sending a proof
+}
+
+// stream of leaves in the contracts tree
+message ContractRange {
+    repeated ContractState state = 1;
+}
+
+// duplicate of GetContractRange. Can introduce a 'type' instead.
+// result is (Classes+, PatriciaRangeProof)*
```

Comment:
The code seems to try and parse the result as a Class -- I don't seem to be able to find the representation of the `(Classes+, PatriciaRangeProof)*` mentioned above.

The same is true for some other messages where the comments and the code don't seem to match.

Reply:
as for snapshot - we won't be using it
as for comments vs code - the spec is still changing and there will be more changes incoming with the sync handlers update; i treat this PR as a WiP that allows the next PR to be smaller

Reply:
we also don't have response streaming yet which those comments indicate